integration-tests/src/test/java/io/opentelemetry/OtlpExporterIntegrationTest.java

```diff
@@ -0,0 +1,355 @@
+import io.opentelemetry.proto.metrics.v1.AggregationTemporality;
+import io.opentelemetry.proto.metrics.v1.InstrumentationLibraryMetrics;
+import io.opentelemetry.proto.metrics.v1.Metric;
+import io.opentelemetry.proto.metrics.v1.NumberDataPoint;
+import io.opentelemetry.proto.metrics.v1.ResourceMetrics;
+import io.opentelemetry.proto.metrics.v1.Sum;
+import io.opentelemetry.proto.trace.v1.InstrumentationLibrarySpans;
+import io.opentelemetry.proto.trace.v1.ResourceSpans;
+import io.opentelemetry.proto.trace.v1.Span.Link;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.export.IntervalMetricReader;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.IdGenerator;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+/**
+ * Integration test to verify OTLP gRPC and HTTP exporters work. Verifies integration with the
+ * OpenTelemetry Collector. The Collector is configured to accept OTLP data over gRPC and HTTP and
+ * to export data over gRPC to a server running in process, allowing assertions to be made against
+ * the data.
+ */
+@Testcontainers(disabledWithoutDocker = true)
+class OtlpExporterIntegrationTest {
+
+  private static final String COLLECTOR_IMAGE = "otel/opentelemetry-collector:0.33.0";
+  private static final Integer COLLECTOR_OTLP_GRPC_PORT = 4317;
+  private static final Integer COLLECTOR_OTLP_HTTP_PORT = 4318;
+  private static final Integer COLLECTOR_HEALTH_CHECK_PORT = 13133;
+  private static final Integer IN_PROCESS_OTLP_GRPC_PORT = 4319;
+  private static final Resource RESOURCE =
+      Resource.getDefault().toBuilder()
+          .put(ResourceAttributes.SERVICE_NAME, "integration test")
+          .build();
+  private static final Logger LOGGER =
+      Logger.getLogger(OtlpExporterIntegrationTest.class.getName());
+
+  static {
+    if (DockerClientFactory.instance().isDockerAvailable()) {
```

Comment:
I think we can do in a BeforeAll to avoid this check (we have it in the `TestContainers` annotation above)

Reply:
That's true. We'll also have to manually start / stop the `GRPC_SERVER` and the `COLLECTOR` container since we'll need to call `exposeHostPorts(..)` after the grpcServer (so that we can use the randomly assigned port) and before the collector is started.